ucm2/Intel/avs/avs_da7219/Google-Atlas-1.0-HiFi.conf

```diff
@@ -0,0 +1,67 @@
+SectionVerb {
+        EnableSequence [
+                cdev "hw:avsda7219"
+                cset "name='Headphone Jack Switch' off"
+                cset "name='Headset Mic Switch' off"
+                cset "name='Playback Digital Volume' 111"
```

Comment:
Ack.

Reply:
In which file should I put BootSequence, it seems to vary and I can't seem to get it to work? Is it supported in "syntax 6"?

I've used ./ucm.py from alsa-tests repo in python/ucm-validator and it seems to suggest following split:

```
$ cat Intel/avs/avs_da7219/avs_da7219.conf                                                                                                                                                                               (git)-[master]-
Syntax 6

SectionUseCase."HiFi" {
        File "/Intel/avs/avs_da7219/avs_da7219-HiFi.conf"
        Comment "Play HiFi quality Music"
}

BootSequence [
        cset "name='Headphone Jack Switch' off"
        cset "name='Headset Mic Switch' off"
        cset "name='Playback Digital Volume' 111"
        cset "name='Playback Digital Switch' 1"
        cset "name='Out DACL Mux' DAIL"
        cset "name='Out DACR Mux' DAIR"
        cset "name='Mixer Out FilterL DACL Switch' 1"
        cset "name='Mixer Out FilterR DACR Switch' 1"
        cset "name='ST Mixer Out FilterL Out FilterL Switch' 1"
        cset "name='ST Mixer Out FilterR Out FilterR Switch' 1"
        cset "name='Mic Volume' 5"
        cset "name='Capture Digital Volume' 111"
        cset "name='Capture Digital Switch' 1"
        cset "name='Mixin Volume' 10"
        cset "name='Mixin Switch' 1"
        cset "name='Mixer In Mic Switch' 1"
        cset "name='Out DAIL Mux' ADC"
        cset "name='Out DAIR Mux' ADC"
        cset "name='Gain Ramp Rate' 1"
]
```

```
$ cat Intel/avs/avs_da7219/avs_da7219-HiFi.conf                                                                                                                                                                          (git)-[master]-
SectionDevice."Headphone" {
        Comment "Headphones"

        Value {
                PlaybackPCM "hw:${CardId},1"
                PlaybackChannels 2
                PlaybackMixerElem "Headphone"
                JackDev "avs_da7219 Headset Jack"
        }
        EnableSequence [
                cset "name='Headphone Jack Switch' on"
                cset "name='Headphone Switch' 1"
        ]
        DisableSequence [
                cset "name='Headphone Jack Switch' off"
                cset "name='Headphone Switch' 0"
        ]
}

SectionDevice."Mic" {
        Comment "Headset Microphone"

        Value {
                CapturePCM "hw:${CardId},1"
                CaptureChannels 2
                CaptureMixerElem "Headset Mic"
                JackDev "avs_da7219 Headset Jack"
        }

        EnableSequence [
                cset "name='Headset Mic Switch' on"
                cset "name='Mic Switch' on"
        ]
        DisableSequence [
                cset "name='Headset Mic Switch' off"
                cset "name='Mic Switch' off"
        ]
}
```

However on test machine controls are not set how BootSequence describes them.

Reply:
Refer to: https://www.alsa-project.org/alsa-doc/alsa-lib/group__ucm__conf.html - `Boot (alsactl)` paragraph. The boot sequence is executed only when the sound state file does not exist to allow users change the default values.

Reply:
The `FixedBootSequence` is not for all values. Also, the values are applied only on boot (usually through `alsa-state` systemd service). You can invoke/force those actions using `alsaucm set _boot ""` or `alsaucm set _fboot ""` commands manually.

Reply:
Well, as a user if I get UCM file my expectation would be that it sets controls as needed for the card to operate properly whether on boot on otherwise. In case of da7219 on this particular device it needs those specific values (card has 83 controls, so I don't expect users to try and set them, so alsactl can do its job remembering them and restoring them on boot), otherwise it won't playback/record. Now you are suggesting that some Values should not be set using either BootSequence or FixedBootSequence, then how they should be set? My assumption is that by providing UCM file controls will be set properly, which worked fine when I started this pull request and was tested on all devices in PR, now I'm trying to follow your suggestions, but they do not seem to work as expected.

Reply:
I would suggest to be familiar with the UCM documentation. In case of `BootSequence` the values should be set when the state file does not exist. The contents of the state file should contain usually similar values unless user changes them for a reason. Usually, default volumes should be handled in this way. For `FixedBootSequence`, values should be always set on boot. I expect that your distro follows the ALSA state file handling. Note that you can always analyze the state file, if control values are preset correctly from UCM and you can start with scratch - remove the state file (usually `/var/lib/alsa/asound.state`, also don't forget to stop `alsa-state` or `alsa-restore` systemd service as first to avoid write the actual state on reboot). Then restart the appropriate service or reboot.

If `FixedBootSequence` does not work, then I would check how your distro is handling the ALSA soundcard state.

Reply:
Ok, removing the file seems to help, although it is not the most user friendly flow from my perspective.

Anyway, one more question, if I want to force specific bit format like S16_LE what is the command, it seems to be missing from documentation, is it "CaptureFormat", or something else?
```
Syntax 6

SectionUseCase."HiFi" {
        File "/Intel/avs/avs_dmic/DMIC-HiFi.conf"
        Comment "Play HiFi quality Music"
}

ValueDefaults {
        CaptureRate 48000
        CaptureChannels 4
        CaptureFormat S16_LE
}
```